net/yggdrasil/files/ygguci

```diff
@@ -0,0 +1,247 @@
+  end)
+  cursor:foreach("yggdrasil", "listen_address", function (s) 
+      table.insert(obj.Listen, s.uri)
+  end)
+  cursor:foreach("yggdrasil", "multicast_interface", function (s) 
+      table.insert(obj.MulticastInterfaces, s.name)
+  end)
+  cursor:foreach("yggdrasil", "allowed_encryption_public_key", function (s) 
+      table.insert(obj.AllowedEncryptionPublicKeys, s.key)
+  end)
+
+  cursor:foreach("yggdrasil", "interface_peer", function (s) 
+      if obj.InterfacePeers[s.interface] == nil then
+        obj.InterfacePeers[s.interface] = {}
+      end
+      table.insert(obj.InterfacePeers[s["interface"]], s.uri)
+  end)
+
+  local node_info = cursor:get_all("yggdrasil", "NodeInfo")
+  for k, v in pairs(node_info) do
+      if string.sub(k, 1, 1) ~= "." then obj.NodeInfo[k] = v end
+  end
+
+  -- session firewall config
+  local session_firewall_config = { "Enable", "AllowFromDirect", "AllowFromRemote", "AlwaysAllowOutbound" }
+  for _, v in pairs(session_firewall_config) do
+      if config["SessionFirewall_"..v] ~= nil then
+          obj.SessionFirewall[v] = to_bool(config["SessionFirewall_"..v])
+      end
+  end
+  cursor:foreach("yggdrasil", "whitelisted_encryption_public_key", function (s) 
+      table.insert(obj.SessionFirewall.WhitelistEncryptionPublicKeys, s.key)
+  end)
+  cursor:foreach("yggdrasil", "blacklisted_encryption_public_key", function (s) 
+      table.insert(obj.SessionFirewall.BlacklistEncryptionPublicKeys, s.key)
+  end)
+  -- /session firewall config
+
+  -- tunnel routing config
+  if config.TunnelRouting_Enable ~= nil then
+      obj.TunnelRouting.Enable = to_bool(config.TunnelRouting_Enable)
+  end
+  cursor:foreach("yggdrasil", "ipv6_remote_subnet", function (s) 
+      obj.TunnelRouting.IPv6RemoteSubnets[s.subnet] = s.key
+  end)
+  cursor:foreach("yggdrasil", "ipv6_local_subnet", function (s) 
+      table.insert(obj.TunnelRouting.IPv6LocalSubnets, s.subnet)
+  end)
+  cursor:foreach("yggdrasil", "ipv4_remote_subnet", function (s) 
+      obj.TunnelRouting.IPv4RemoteSubnets[s.subnet] = s.key
+  end)
+  cursor:foreach("yggdrasil", "ipv4_local_subnet", function (s) 
+      table.insert(obj.TunnelRouting.IPv4LocalSubnets, s.subnet)
+  end)
+  -- /tunnel routing config
+
+  if config.SwitchOptions_MaxTotalQueueSize ~= nil then
+      obj.SwitchOptions.MaxTotalQueueSize = tonumber(config.SwitchOptions_MaxTotalQueueSize)
+  end
+
+  return obj
+end
+
+--- Parse and save updated configuration from JSON input
+--
+-- Transforms general settings into UCI sections, and replaces the UCI config's
+-- contents with them.
+-- @param table JSON input
+-- @return Boolean whether saving succeeded
+function UCI.set(obj)
+  local cursor = uci.cursor()
+
+  for i, section in pairs(cursor:get_all("yggdrasil")) do
+    cursor:delete("yggdrasil", section[".name"])
+  end
+
+
+  cursor:set("yggdrasil", "yggdrasil", "yggdrasil")
+  cursor:set("yggdrasil", "yggdrasil", "EncryptionPublicKey", obj.EncryptionPublicKey) 
```

Comment:
Can we please lowercase all UCI options names? All uci options (as far as i can tell) should be lower case.

Reply:
Do you think lowercase option names is a better common practice? FYI /etc/config/dropbear also has CamelCase option names. Those settings are direct mappings for the same keys in yggdrasil.conf, so I thought it could make more sense.

Reply:
Could flip a coin either way.